cloud/shared/bin/doctor_impl

```diff
@@ -0,0 +1,221 @@
+        """
+
+class AwsCliCheck(Check):
+    remedy = """
+        Need to install the AWS CLI tool.
+        https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+    """
+
+    def name(self):
+        return "AWS CLI"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() not in ['aws', 'azure']
+
+    def is_ok(self):
+        return self._check_command_succeeds(["aws", "help"])
+
+class AzureCliCheck(Check):
+    remedy = """
+        Need to install the Azure CLI tool.
+        https://docs.microsoft.com/en-us/cli/azure/install-azure-cli
+    """
+
+    def name(self):
+        return "Azure CLI"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() != "azure"
+
+    def is_ok(self):
+        return self._check_command_succeeds(["az"])
+
+class AwsLoginCheck(Check):
+    remedy = """
+        Need to configure AWS CLI tool. Run `aws configure` and enter credentials
+        or for other options visit
+        https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html
+    """
+
+    def name(self):
+        return "AWS login"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() not in ['aws', 'azure']
+
+    def is_ok(self):
+        output = self._get_command_output(["aws", "configure", "list-profiles"])
+
+        # the default profile will be present if they are logged in
+        return output and 'default' in output.stdout.decode('ascii')
+
+class AzureLoginCheck(Check):
+    remedy = """
+        Need to authenticate with the Azure CLI tool.
+        https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli
+    """
+
+    def name(self):
+        return "Azure login"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() != "azure"
+
+    def is_ok(self):
+        return self._check_command_succeeds(["az", "ad", "signed-in-user", "show"])
+
+checks = [
+    DockerCheck(),
+    TerraformCheck(),
+    JavaCheck(),
+    KeytoolCheck(),
+    AwsCliCheck(),
+    AzureCliCheck(),
+    AwsLoginCheck(),
+    AzureLoginCheck()
+]
+
+problems = []
+for check in checks:
+    if check.should_skip():
+        continue
+
+    print(check.name().ljust(30, "."), end="", flush=True)
+
+    if check.is_ok():
+        print("OK")
+        continue
+
+    print("FAILED")
+    problems.append(check)
+
+if len(problems) == 0:
+    print("All checks passed!")
+    exit(0)
+
+print("\nSome checks failed, please do the fix for each failure:")
+
+for failed_check in problems:
+    print(f"\n{failed_check.name()}")
+
+    if is_linux:
+        remedy = failed_check.remedy_linux()
+    elif is_mac:
+        remedy = failed_check.remedy_mac()
+    else:
+        raise RuntimeError("Only Linux and macOS are supported.")
```

Comment:
we do technically check this earlier but i don't mind the repeated check

Reply:
yeah, if we end up supporting BSD or Solaris or something down the line this error should hopefully make it more obvious that the logic here needs to be updated as well